src/core/CMakeLists.txt

```diff
@@ -1112,9 +1113,7 @@ IF (WIN32)
   TARGET_LINK_LIBRARIES(qgis_core wsock32 ${SETUPAPI_LIBRARY} DbgHelp)
 ENDIF (WIN32)
 
-IF(APPLE)
-  TARGET_LINK_LIBRARIES(qgis_core "-framework CoreFoundation -framework IOKit")
-ENDIF(APPLE)
+TARGET_LINK_LIBRARIES(qgis_core qgis_native)
```

Comment:
@dakcarto any ideas here? It builds fine for Linux without this, but fails on osx. Is it something to do with the native osx cocoa stuff?